pkg/serverless/flush/strategy.go

```diff
@@ -0,0 +1,104 @@
+package flush
+
+import (
+	"time"
+)
+
+// Strategy is deciding whether the data should be flushed or not at the given moment.
+type Strategy interface {
+	String() string
+	ShouldFlush(moment Moment, t time.Time) bool
+}
+
+// Moment represents at which moment we're asking the flush Strategy if we
+// should flush or not.
+// Note that there is no entry for the shutdown of the environment because we always
+// flush in this situation.
+type Moment string
+
+const (
+	// Starting is used to represent the moment the function is starting because
+	// it has been invoked.
+	Starting Moment = "starting"
+	// Stopping is used to represent the moment right after the function has finished
+	// its execution.
+	Stopping Moment = "stopping"
+	// Running is used to indicate that the function is still running.
+	// Running Moment = "running"
```

Comment:
Is `Running` supposed to be commented out here?

Reply:
May reappear with further changes but indeed, it is unused in this first implementation. Removing it 👍